src/HorusStudio.Maui.MaterialDesignControls/Controls/MaterialProgressIndicator.cs

```diff
@@ -8,153 +7,325 @@ public enum MaterialProgressIndicatorType
         Circular, Linear
     }
 
+    /// <summary>
+    /// A progress indicator <see cref="View" /> that show the status of a process and follows Material Design Guidelines.
+    /// </summary>
     public class MaterialProgressIndicator : ContentView
     {
         #region Attributes and Properties
 
-        private bool _initialized = false;
+        private readonly static MaterialProgressIndicatorType DefaultProgressIndicatorType = MaterialProgressIndicatorType.Circular;
+        private readonly static Color DefaultIndicatorColor = MaterialLightTheme.Primary;
```

Comment:
we should use extension to determine current theme i.e. `new AppThemeBindingExtension { Light = MaterialLightTheme.Primary, Dark = MaterialDarkTheme.Primary }.GetValueForCurrentTheme<Color>();`